### ISSUE TITLE
Bug: [await-thenable] Also flag unknown

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8301]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3hkmBc8ArigNYpIDuKA3AFAAUUaAnimPPQJTwC8AfPADetePCgUoASyw5MtAL6cu1IA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQwHd3K78ALRE3YAjJOiiJo0APbRI4MAF8QSoA&tsconfig=&tokens=false

### Repro Code

```TypeScript
declare const test: unknown;
(async () => {
  await test
})();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/await-thenable": "error"
  },
};
```

### tsconfig

```jsonc
{}
```

### Expected Result

There's no guarantee that an `unknown`-typed value is a Thenable. I'd expect the rule to report a complaint in this case.

### Actual Result

No complaint.

### Additional Info

`await-thenable` was first implemented in #192 back in 2019. TypeScript had previously released `unknown` in [TypeScript 3.0](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0/#the-unknown-type) less than a year earlier. 

Not detecting `unknown` was noted by @Tjstretchalot in https://github.com/typescript-eslint/typescript-eslint/pull/8094#issuecomment-1908627372.

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages%2Feslint-plugin%2Fsrc%2Frules%2Fawait-thenable.ts#L40-L42

The rule intentionally ignores any and unknown because of your stated reason -- or rather, it inverted - there's no guarantee that an any/unknown ***isn't*** a promise.

IMO it's incorrect for the rule to flag this case because the thing might be a promise.

Put another way - if we flag any/unknown:
- and it IS a promise, then removing the `await` is the incorrect outcome as removing it will break the code (massive negative)
- and it IS NOT a promise, then removing the `await` is correct, but likely won't change code behaviour at all (small positive)

Looking at it this way - by reporting a user uwll either completely and unknowingly break their code, or they'll clean up a mostly stylistic concern... To me that screams "do not report".

**Comment by JoshuaKGoldberg** [comment]
Hmm, that makes sense. I don't feel particularly strongly one way or another tbh. Users haven't been complaining (to my knowledge) at any rate.

cc @Tjstretchalot from https://github.com/typescript-eslint/typescript-eslint/pull/8094#issuecomment-1912453518 - sounds like we can just close this out as wontfix and go with what you'd originally proposed for #8094. Thoughts? 😄
